### PR TITLE
OKTA-682407: Fixed broken navigation after first load and removed dynamic anchors in URL

### PR DESCRIPTION
**Pull request by ishan-krishna-okta**
## Description:
- **What's changed?** 
   - Fixed broken navigation which did not highlight the active header on scroll in the right side nav.
   - Removed dynamic anchors in the URL on scroll
- **Is this PR related to a Monolith release?** No

### Resolves:

* [OKTA-682407](https://oktainc.atlassian.net/browse/OKTA-682407)

### CODE REVIEW

**Comment by ishan-krishna-okta** [comment]
Preview URL - https://659f8f7e89b44735d2278abe--reverent-murdock-829d24.netlify.app

**Review by emikhasyak-okta — COMMENTED** [review]
I left a couple of comments. For the functionality, we need to have `resize` event working. For disabling hash syncing in address bar - we should clean as much as possible unused code.

File: packages/@okta/vuepress-theme-prose/components/OnThisPage.vue [hunk 1]
```diff
@@ -78,12 +79,25 @@ export default {
...
+    this.setAnchors(this.getOnThisPageAnchors());
+    this.setActiveAnchor();
+    window.addEventListener("scroll", this.setActiveAnchor);
+  },
+  updated() {
+    if (!this.isCalledOnceFromUpdated) {
+      // Sometimes anchors are not set during the mounting phase. Hence, we need to set the anchors again 
+      // in the updated hook and we only need to do it once, hence, the isCalledOnceFromUpdated condition.
+      // Adding a setTimeout as due to some reason this was not working in the preview build but was working
+      // locally. Adding a setTimeout fixes the issue in the preview build.
```
[the remark this answers is not in the document]
Comment (ishan-krishna-okta) [reply packages/@okta/vuepress-theme-prose/components/OnThisPage.vue:90]: Tried putting the setTimeout in mounted hook. It is not solving the issue. We still need to put it in `updated` hook.

File: packages/@okta/vuepress-theme-prose/mixins/AnchorHistory.vue [hunk 2]
```diff
@@ -86,11 +86,6 @@ export default {
 
       if (decodeURIComponent(this.$route.hash) !== decodeURIComponent(anchor)) {
         this.$vuepress.$set("disableScrollBehavior", true);
-        this.$router.replace(anchor, () => {
```
[the remark this answers is not in the document]
Comment (ishan-krishna-okta) [reply packages/@okta/vuepress-theme-prose/mixins/AnchorHistory.vue:89]: Great catch. I have removed the unnecessary code now.

File: packages/@okta/vuepress-theme-prose/components/OnThisPage.vue [hunk 3]
```diff
@@ -78,12 +79,25 @@ export default {
   mounted() {
     window.addEventListener("load", () => {
       this.$nextTick(() => {
-        this.setAnchors(this.getOnThisPageAnchors());
-        this.setActiveAnchor();
-        window.addEventListener("scroll", this.setActiveAnchor);
         window.addEventListener("resize", this.updateAnchors);
```
[the remark this answers is not in the document]
Comment (ishan-krishna-okta) [reply packages/@okta/vuepress-theme-prose/components/OnThisPage.vue]: Updated.

**Comment by ishan-krishna-okta** [comment]
Updated preview URL - https://65a0dfcec51a3b8771cdbad6--reverent-murdock-829d24.netlify.app/

**Review by emikhasyak-okta — APPROVED** [review]
Perfect! :shipit:

**Pull request merged**